templates/generic/service_and_deployment.yaml

```diff
@@ -0,0 +1,156 @@
+{{- $geoserverName := (include "geoserver.fullname" .) -}}
+{{- $geoserverLabels := (include "geoserver.labels" .) -}}
+{{- $geoserverSelectorLabels := (include "geoserver.selectorLabels" .) -}}
+{{- $geoserverServiceAccountName := (include "geoserver.serviceAccountName" .) }}
+{{- $chart := .Chart -}}
+{{- $shaConfigmap := (include (print $.Template.BasePath "/config/configmap.yaml") . | sha256sum) -}}
+
+{{- $values := .Values -}}
+
+{{- range $serviceName, $serviceDefinition := $values.geoserver.services }}
+---
+# service {{ $serviceName }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ $geoserverName }}-{{ $serviceName }}
+  labels:
+    {{- $geoserverLabels | nindent 4 }}
+    gs-service: {{ $serviceName }}
+spec:
+  type: {{ $serviceDefinition.service.type }}
+  ports:
+    - port: {{ $serviceDefinition.service.port }}
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    {{ $geoserverSelectorLabels | nindent 4  }}
+    gs-service: {{ $serviceName }}
```

Comment:
The standard name is `app.kubernetes.io/component` :-)

Reply:
@sbrunner  you mean : instead of `gs-service` ?

Reply:
yes :-)